### PR TITLE
[frontend] add error msgs + loader

### PR DESCRIPTION
**Pull request by darren-wangg**
## Description
- relevant slack threads: https://aptos-org.slack.com/archives/C03RGDZ658X/p1728071717317759, https://aptos-org.slack.com/archives/C03RGDZ658X/p1728071839845759
- there were not very descriptive error messages for what was failing in proposal data fetching
  - i.e. `Proposal not found` for everything
- also, `Proposal not found` would flash while proposal data was fetching

Added:
- some error messages for metadata errors
- loading state when fetching proposals

---
Before:

https://github.com/user-attachments/assets/49bf0ddd-0e0f-439b-93fe-27ddd3895337

After:

https://github.com/user-attachments/assets/35a258d8-4e57-4a69-aa19-7d9e691357f4

### CODE REVIEW

File: src/api/hooks/useGetProposal.tsx [hunk 1]
```diff
@@ -56,36 +57,50 @@ const fetchProposalMetadata = async (
...
+  // validate metadata
   const metadata_hash = proposalData.metadata.data.find(
     (metadata) => metadata.key === "metadata_hash",
   )!.value;
-
   const hash = sha3_256(metadataText);
-  if (hex_to_string(metadata_hash) !== hash) return null;
+  if (hex_to_string(metadata_hash) !== hash) {
+    return {
+      errorMessage: "Metadata is invalid",
```
Comment (darren-wangg) [inline src/api/hooks/useGetProposal.tsx]: what should this error text be?

Comment (darren-wangg) [reply src/api/hooks/useGetProposal.tsx]: this is what was causing the error in the slack threads

Comment (sherry-x) [reply src/api/hooks/useGetProposal.tsx]: "Metadata hash mismatch" is probably more accurate here

Comment (darren-wangg) [reply src/api/hooks/useGetProposal.tsx]: updated

**Review by sherry-x — APPROVED** [review]

**Pull request merged**